luaui/Widgets/api_resource_spot_builder.lua

```diff
@@ -142,122 +142,212 @@ local function IsSpotOccupied(spot)
 	return false
 end
 
+
+local function resourceSpotHasExistingExtractorCommand(x, z, builders)
+	for i=1, #builders do
+		local queue = Spring.GetCommandQueue(builders[i], 100)
+		for j=1, #queue do
+			local command = queue[j]
+			local id = -command.id
+			if(mexBuildings[id] or geoBuildings[id]) then
+				local dist = math.sqrt(distance2dSquared(x, z, command.params[1], command.params[3]))
```

Comment:
Here we can avoid the sqrt by have `4 * Game_extractorRadiusSq` in the check below.